### PR TITLE
Watson/fast g2 check

### PR DESCRIPTION
**Pull request by wbl**

### CODE REVIEW

**Comment by tanyav2** [comment]
We should add @bwesterb to these PRs since he's also part of the working group

**Comment by wbl** [comment]
get him set up in the cf org: i can't tag him right now, suspect that's why

File: ecc/bls12381/constants.go [hunk 1]
```diff
@@ -35,6 +44,17 @@ func err(e error) {
 	}
 }
 
+func ratioKummer(t *ff.Fp12) (*ff.Fp2, error) {
```
Comment (armfazh) [inline ecc/bls12381/constants.go:47]:
```suggestion
// ratioKummer returns t^(1-p) = t / Frob(t). 
func ratioKummer(t *ff.Fp12) (*ff.Fp2, error) {
```

File: ecc/bls12381/constants.go [hunk 2]
```diff
@@ -27,6 +31,11 @@ var (
 		beta0 ff.Fp    // beta0 = F(2)^(2*(p-1)/3) where F = GF(p).
 		beta1 ff.Fp    // beta1 = F(2)^(1*(p-1)/3) where F = GF(p).
 	}
+	g2PsiCoeff struct {
+		minusZ [16]byte // -z, the BLS12 parameter
+		alpha  ff.Fp2   // alpha = t^2/\sigma(t^2) where t=w
+		beta   ff.Fp2   // beta = t^3/\sigma(t^3) where t=w
```
Comment (armfazh) [inline ecc/bls12381/constants.go:37]:
```suggestion
		alpha  ff.Fp2   // alpha = t^2/Frob(t^2) where t=w
		beta   ff.Fp2   // beta = t^3/Frob(t^3) where t=w
```

**Review by armfazh — APPROVED** [review]
a couple of nits.

File: ecc/bls12381/g2.go [hunk 3]
```diff
@@ -59,9 +59,39 @@ func (g *G2) cmov(P *G2, b int) {
 	(&g.y).CMov(&g.y, &P.y, b)
 	(&g.z).CMov(&g.z, &P.z, b)
 }
+func (g *G2) IsRTorsionSlow() bool {
```
Comment (armfazh) [inline ecc/bls12381/g2.go:62]: not needed, except in testing

**Pull request merged**